libraries/botbuilder-dialogs-adaptive/src/entityAssignments.ts

```diff
@@ -11,28 +11,31 @@ import { EntityAssignment } from './entityAssignment';
 
 const events = 'this.events';
 
+export interface EntityAssignmentsConfiguration {
+    assignments: EntityAssignment[];
+}
+
 /**
  * Tracks entity related events to surface.
  * @remarks When processing entities possible ambiguities are identified and when resolved they turn into assign events.
  * This tracking persists across multiple input utterances.
  */
-export class EntityAssignments {
-    /**
-     * Queue of entity assignments.
-     */
-    public assignments: Partial<EntityAssignment>[] = [];
+export class EntityAssignments implements EntityAssignmentsConfiguration {
+    constructor(public assignments: EntityAssignment[] = []) {}
 
     /**
      * Read entity event queue from memory.
      * @param actionContext Memory context.
      */
     public static read(actionContext: ActionContext): EntityAssignments {
-        let queues: EntityAssignments = actionContext.state.getValue(events);
-        if (!queues) {
-            queues = new EntityAssignments();
+        let queuesObject: EntityAssignments = actionContext.state.getValue(events);
```

Comment:
I think `getValue` supports a second `defaultValue` parameter which would make the `if` check unnecessary.